File: library/src/matrix_energy/lib/type/MEItem.java
package matrix_energy.lib.type;

import arc.graphics.Color;
import arc.struct.Seq;
import arc.util.Nullable;
import mindustry.type.Item;

/**
 * @author DG
 */
public class MEItem extends Item {
    public static Seq<MEItem> all = new Seq<>();
    public @Nullable Item imitateItemIcon;

    public MEItem(String name) {
        this(name, Color.black.cpy(), name, null);
    }

    public MEItem(String name, Color color, String localizedName, Item imitateItemIcon) {
        super(name, color);
        this.localizedName = localizedName;
        this.imitateItemIcon = imitateItemIcon;
        all.add(this);

    }

    public MEItem(String name, Color color) {
        this(name, color, name, null);
    }

    public MEItem(String name, Color color, String localizedName) {
        this(name, color, localizedName, null);
    }

    @Override
    public void loadIcon() {
        super.loadIcon();
        if (imitateItemIcon != null) {
            fullIcon = imitateItemIcon.fullIcon != null ? imitateItemIcon.fullIcon : fullIcon;
            uiIcon = imitateItemIcon.uiIcon != null ? imitateItemIcon.uiIcon : uiIcon;
        }
    }
}

File: library/src/matrix_energy/lib/type/item/MEItem.kt
package matrix_energy.lib.type.item

import arc.graphics.Color
import arc.util.io.Reads
import arc.util.io.Writes
import matrix_energy.lib.MatrixEnergyInterfaces.MatrixEnergyItem
import mindustry.type.Item

/**
 * @author DG
 */
open class MEItem @JvmOverloads constructor(
    name: String,
    color: Color = Color.black.cpy()
) : Item(name, color), MatrixEnergyItem {
    
    constructor(name: String, localizedName: String, color: Color = Color.black.cpy()): this(name, color){
        this.localizedName = localizedName
    }

    init {
        this.localizedName = localizedName
//        all.add(this)
    }

    fun read(read: Reads, revision: Byte) {
    }

    fun write(write: Writes) {
    }

    override fun setStats() {
        super.setStats()
    }
}
